src/lib/CrsfProtocol/crsf_protocol.h

```diff
@@ -364,7 +364,8 @@ typedef struct crsfPayloadLinkstatistics_s
     uint8_t active_antenna;
     uint8_t rf_Mode;
     uint8_t uplink_TX_Power;
-    uint8_t downlink_RSSI;
+    uint8_t downlink_RSSI_1;
+    uint8_t downlink_RSSI_2;
```

Comment:
You can't just add a field to this structure I'm afraid, as this structure defines the CRSF on-the-wire packet. With this PR, the LinkStatistics going out as CRSF are broken. You'll have to store the data somewhere else.

Reply:
Not thrilled about doing this considering how things might later expect the struct to be a certain size, but the way our code is written now, this works (subclassing the wire struct to add a field). Also removes volatile because that is not needed and shaves off 84 bytes of instructions. Can push if you like.
```diff
diff --git a/src/lib/CrsfProtocol/crsf_protocol.h b/src/lib/CrsfProtocol/crsf_protocol.h
index 5a4c080e..077b41fd 100644
--- a/src/lib/CrsfProtocol/crsf_protocol.h
+++ b/src/lib/CrsfProtocol/crsf_protocol.h
@@ -364,12 +364,15 @@ typedef struct crsfPayloadLinkstatistics_s
     uint8_t active_antenna;
     uint8_t rf_Mode;
     uint8_t uplink_TX_Power;
-    uint8_t downlink_RSSI;
+    uint8_t downlink_RSSI_1;
     uint8_t downlink_Link_quality;
     int8_t downlink_SNR;
-} crsfLinkStatistics_t;
+} PACKED crsfLinkStatistics_t;

-typedef struct crsfPayloadLinkstatistics_s crsfLinkStatistics_t;
+typedef struct elrsLinkStatistics_s : crsfLinkStatistics_t
+{
+    uint8_t downlink_RSSI_2;
+} PACKED elrsLinkStatistics_t;

 // typedef struct crsfOpenTXsyncFrame_s
 // {
diff --git a/src/lib/Handset/CRSF.cpp b/src/lib/Handset/CRSF.cpp
index 8cdaac03..ea1eab58 100644
--- a/src/lib/Handset/CRSF.cpp
+++ b/src/lib/Handset/CRSF.cpp
@@ -3,7 +3,7 @@
 #include "common.h"
 #include "FIFO.h"

-volatile crsfPayloadLinkstatistics_s CRSF::LinkStatistics;
+elrsLinkStatistics_t CRSF::LinkStatistics;
 GENERIC_CRC8 crsf_crc(CRSF_CRC_POLY);

 uint8_t CRSF::MspData[ELRS_MSP_BUFFER] = {0};
diff --git a/src/lib/Handset/CRSF.h b/src/lib/Handset/CRSF.h
index 01eb208b..4bf3a2f3 100644
--- a/src/lib/Handset/CRSF.h
+++ b/src/lib/Handset/CRSF.h
@@ -9,7 +9,7 @@
 class CRSF
 {
 public:
-    static volatile crsfPayloadLinkstatistics_s LinkStatistics; // Link Statistics Stored as Struct
+    static elrsLinkStatistics_t LinkStatistics; // Link Statistics Stored as Struct

     static void GetMspMessage(uint8_t **data, uint8_t *len);
     static void UnlockMspMessage();
diff --git a/src/src/tx_main.cpp b/src/src/tx_main.cpp
index 84b616a3..821ff122 100644
--- a/src/src/tx_main.cpp
+++ b/src/src/tx_main.cpp
@@ -212,7 +212,7 @@ bool ICACHE_RAM_ATTR ProcessTLMpacket(SX12xxDriverCommon::rx_status const status

   Radio.GetLastPacketStats();
   CRSF::LinkStatistics.downlink_SNR = SNR_DESCALE(Radio.LastPacketSNRRaw);
-  CRSF::LinkStatistics.downlink_RSSI = Radio.LastPacketRSSI;
+  CRSF::LinkStatistics.downlink_RSSI_1 = Radio.LastPacketRSSI;

   // Full res mode
   if (OtaIsFullRes)
```

Reply:
>Can push these changes (and all the other ones I suggested) if you agree.

I just got sometime to sit down and go through all of this properly... then read this!

YES PLEASE, PUSH ALL THE THINGS!

Reply:
> You can't just add a field to this structure I'm afraid, as this structure defines the CRSF on-the-wire packet. With this PR, the LinkStatistics going out as CRSF are broken. You'll have to store the data somewhere else.

Yep, I totally fucked this.  I was to busy looking at OTA packets :sweat_smile:

Reply:
> Yep, I totally fucked this. I was to busy looking at OTA packets 😅

That's what I thought. Jye checking out the back of his module admiring how cool it looks, while on the other side the linkstats are displaying all weird on the handset 😄